torchquad/tests/gradient_test.py

```diff
@@ -0,0 +1,62 @@
+import sys
+
+sys.path.append("../")
+
+import torch
+
+from integration.vegas import VEGAS
+from integration.monte_carlo import MonteCarlo
+from integration.trapezoid import Trapezoid
+from integration.simpson import Simpson
+from integration.boole import Boole
+
+from utils.enable_cuda import enable_cuda
+from utils.set_precision import set_precision
+
+
+def test_function(x):
+    """V shaped test function.
+    Gradient in positive x should be 2,
+    Gradient in negative x should be -2
+    for -1 to 1 domain."""
+    return 2 * torch.abs(x)
+
+
+def test_gradients():
+    """Tests that the implemented integrators
+    maintain torch gradients and they are consistent and correct"""
+    enable_cuda()
+    set_precision("double")
+
+    N = 99997
+    result_tolerence = 1e-2
+    gradient_tolerance = 1e-2
+    torch.manual_seed(0)  # we have to seed torch to get reproducible results
+
+    # Define integrators
+    integrators = [MonteCarlo(), Trapezoid(), Simpson(), Boole(), VEGAS()]
+    for integrator in integrators:
+        # Compute integral
```

Comment:
Can you print what method are you testing? Otherwise, it is difficult to get what method is failing in case of failure.

Reply:
Done